public/app/features/explore/Logs/Logs.tsx

```diff
@@ -332,6 +353,33 @@ class UnthemedLogs extends PureComponent<Props, State> {
     };
   };
 
+  onPermalinkClick = async (row: LogRowModel) => {
+    // get explore state, add log-row-id and make timerange absolute
+    const urlState = getUrlStateFromPaneState(getState().explore.panes[this.props.exploreId]!);
+    urlState.panelsState = { ...this.props.panelState, logs: { id: row.uid } };
+    urlState.range = {
+      from: new Date(this.props.absoluteRange.from).toISOString(),
+      to: new Date(this.props.absoluteRange.to).toISOString(),
+    };
+
+    // append changed urlState to baseUrl
+    const serializedState = serializeStateToUrlParam(urlState);
+    const baseUrl = /.*(?=\/explore)/.exec(`${window.location.href}`)![0];
```

Comment:
I think we could use `generateInternalHref` or  `locationUtil.assureBaseUrl` here instead of regex.

Reply:
Actually I used this solution from https://github.com/grafana/grafana/blob/b5a2c3c7f50b0735a21a533d50d3fa33a5f57faa/public/app/core/utils/richHistory.ts#LL140C1-L142C82.

Based on this Slack convo Explore might add functionality that we can use instead: https://raintank-corp.slack.com/archives/C02TXB92G6T/p1686659940505149?thread_ts=1686656184.982269&cid=C02TXB92G6T

Do you think I should change this now, or do you think it can wait?

Reply:
In this case, it is fine to wait until we have a util function from explore